src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -57,6 +57,12 @@ public override void Dispose()
                     _sslStream.Dispose();
                     _sslStream = null;
                 }
+
+                if (_tcpStream != null)
+                {
+                    _tcpStream.Dispose();
```

Comment:
Looks like the ctor stores _tcpStream into _stream.  Do we need to also null out _stream?

Also looks like most access to _tcpStream and _stream is done under the same lock used here in Dispose, but there are a few cases where _stream is set to either _sslStream or _tcpStream not under the lock (or at least not in that method, maybe it's held by a caller)... is that ok?